server/service/src/processors/transfer/requisition/create_response_requisition.rs

```diff
@@ -126,19 +126,9 @@ impl AuthorisationFields {
         }
         // if not "shouldAuthoriseResponseRequisition" preference  return Auth::default();
 
-        // TODO this would only work with a datafile where master list name matches supplier name
-        // program_id would ideally transfer from request requisition, program_id, this is a hacky fix for now
-        let program_id = MasterListRepository::new(connection)
-            .query_by_filter(MasterListFilter::new().name(SimpleStringFilter::equal_to(
-                &request_requisition.name_row.name,
-            )))
-            .unwrap()[0]
-            .id
-            .clone();
-
         Ok(AuthorisationFields {
             approval_status: Some(RequisitionRowApprovalStatus::Pending),
-            program_id: Some(program_id),
+            program_id: request_requisition.requisition_row.program_id.clone(),
```

Comment:
I was hoping we can reduce this logic to: https://github.com/openmsupply/open-msupply/compare/1504-program-requisitions-program-data-in-requisitions...1504-Approval-status-example